docs/src/_reference/plugin-definition-syntax.md

```diff
@@ -0,0 +1,399 @@
+    description: Create an admin user.
+  ui:
+    args: webserver
+    description: Start the Airflow webserver.
+```
+
+### `commands.<command_name>.args`
+
+Command line arguments for the command.
+
+### `commands.<command_name>.description`
+
+Friendly description of the command.
+
+### `commands.<command_name>.container_spec`
+
+The container specification to use for the command.
+
+Example:
+
+```yaml
+- name: dbt
+  pip_url: dbt-core~=1.0.1 dbt-postgres~=1.0.1
+  commands:
+    compile:
+      args: compile
+      container_spec:
+        command: compile
+        image: ghcr.io/dbt-labs/dbt-postgres:latest
+        env:
+          DBT_PROFILES_DIR: /usr/app/profile/
+        volumes:
+        - "$MELTANO_PROJECT_ROOT/transform/:/usr/app/"
+    docs-generate:
+      args: docs generate
+      container_spec:
+        command: docs generate
+        image: ghcr.io/dbt-labs/dbt-postgres:latest
+        env:
+          DBT_PROFILES_DIR: /usr/app/profile/
+        volumes:
+         - "$MELTANO_PROJECT_ROOT/transform/:/usr/app/"
+    docs-serve:
+      args: docs serve
+      container_spec:
+        command: docs serve --no-browser
+        image: ghcr.io/dbt-labs/dbt-postgres:latest
+        env:
+          DBT_PROFILES_DIR: /usr/app/profile/
+        volumes:
+        - "$MELTANO_PROJECT_ROOT/transform/:/usr/app/"
+        ports:
+          "8080": "8080/tcp"
+```
+
+Use this with [`meltano invoke`](/reference/command-line-interface#containerized-commands) to run commands in a container.
+
+#### `commands.<command_name>.container_spec.image`
+
+The Docker image to use for the command.
+
+#### `commands.<command_name>.container_spec.command`
+
+The command to run in the container.
+
+#### `commands.<command_name>.container_spec.entrypoint`
+
+The container entrypoint to use for the command.
+
+#### `commands.<command_name>.container_spec.ports`
+
+Mapping of host ports to container ports.
+
+#### `commands.<command_name>.container_spec.volumes`
+
+An array of host volumes to mount in the container.
+
+#### `commands.<command_name>.container_spec.env`
+
+A mapping of environment variables to set in the container.
+
+## `settings`
+
+Each plugin variant in Meltano Hub has a `settings` property. Nested under this property are a variable amount of individual settings. In the Meltano UI these settings are parsed to generate a configuration form. To improve the UX of this form, each setting has a number of optional properties.
+
+Example:
+
+```yaml
+settings:
+- name: core.dags_folder
+  label: DAGs Folder
+  value: $MELTANO_PROJECT_ROOT/orchestrate/dags
+  env: AIRFLOW__CORE__DAGS_FOLDER
+- name: core.plugins_folder
+  label: Plugins Folder
+  value: $MELTANO_PROJECT_ROOT/orchestrate/plugins
+  env: AIRFLOW__CORE__PLUGINS_FOLDER
+```
+
+### Protected settings
+
+Until role-based access control is implemented in Meltano, we need to prevent user editing of certain settings from the UI.
+
```

Comment:
I see that this was copy-pasted from the original source material but it looks out of place here and I think it can be deleted.

@edgarrmondragon, wdyt?

Reply:
@aaronsteers agree on deleting it. We already have the `settings[*].protected` section.

Reply:
Deleting it makes sense to me.